### PR TITLE
Performance monitoring config

### PR DESCRIPTION
**Pull request by JasonDing0401**
This is for Skyplane metrics collection feature. A design document can be found [here](https://docs.google.com/document/d/1tiryML0legeSrjJKcsjhGxO839dy-PBAAuY_sob1Jog/edit#heading=h.mxqj8tx7jcs1).

### CODE REVIEW

File: skyplane/__init__.py [hunk 1]
```diff
@@ -18,6 +20,7 @@
 azure_config_path = config_root / "azure_config"
 azure_sku_path = config_root / "azure_sku_mapping"
 gcp_config_path = config_root / "gcp_config"
+clientid_config_path = config_root / "clientid_config"
```
Comment (parasj) [inline skyplane/__init__.py:23]: Can we just include this directly into SkyplaneConfig?

File: skyplane/_private/usage_constants.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
+USAGE_STATS_ENABLED_ENV_VAR = "SKYPLANE_USAGE_STATS_ENABLED"
```
Comment (parasj) [inline skyplane/_private/usage_constants.py:1]: What is the purpose of putting this under _private? It's maybe better to put these files under a `skyplane.cli.usage` directory.

File: skyplane/cli/cli.py [hunk 3]
```diff
@@ -163,15 +168,27 @@ def error_local():
...
             aws_instance_class=cloud_config.get_flag("aws_instance_class"),
             azure_instance_class=cloud_config.get_flag("azure_instance_class"),
             gcp_instance_class=cloud_config.get_flag("gcp_instance_class"),
             gcp_use_premium_network=cloud_config.get_flag("gcp_use_premium_network"),
             multipart_enabled=multipart,
             multipart_max_chunk_size_mb=cloud_config.get_flag("multipart_max_chunk_size_mb"),
         )
-        return 0 if stats["success"] else 1
+
+        if cloud_config.get_flag("verify_checksums"):
```
Comment (parasj) [inline skyplane/cli/cli.py:179]: Is this a bad merge? I think this change is on main now.

File: skyplane/cli/cli.py [hunk 4]
```diff
@@ -413,6 +443,45 @@ def init(
...
     cloud_config.to_config_file(config_path)
     typer.secho(f"\nConfig file saved to {config_path}", fg="green")
+
+    # Set metrics collection by default
+    usage_stats.show_usage_stats_prompt()
+    return 0
+
+
+@app.command()
+def metrics(
```
Comment (parasj) [inline skyplane/cli/cli.py:453]: It's minor but can we potentially fold this into the `skyplane config` CLI as a regular flag like `multipart_enabled`?

File: skyplane/replicate/replicator_client.py [hunk 5]
```diff
@@ -686,42 +702,23 @@ def fn(s: Server):
                     do_parallel(fn, self.bound_nodes.values(), n=-1)
                     progress.update(cleanup_task, description=": Shutting down gateways")
 
-    def verify_transfer(self, job: ReplicationJob):
+    @staticmethod
+    def verify_transfer_prefix(job: ReplicationJob, dest_prefix: str):
```
Comment (parasj) [inline skyplane/replicate/replicator_client.py:706]: Again, this looks like a bad merge.

**Review by parasj — CHANGES_REQUESTED** [review]
PR looks mostly good but I think we can simplify the code by removing the `skyplane metrics` command and then rolling that logic into the normal config flag flow. Otherwise overall design looks great! A minor note is that I think there are some elements of a potentially bad merge. Try to merge/rebase onto main and see if it fixes some of the issues.

File: skyplane/_private/usage_constants.py (hunk 2 above)
Comment (JasonDing0401) [reply skyplane/_private/usage_constants.py:1]: `_private/` is meant for storing user private data. Ray does so [here](https://github.com/ray-project/ray/tree/b52cd964cb1d3696cb3c2bf43f5984bc49eb13e8/python/ray/_private/usage). But I can also move these files to `skyplane.cli.usage`.

File: skyplane/__init__.py (hunk 1 above)
Comment (JasonDing0401) [reply skyplane/__init__.py:23]: I forgot to delete it. `clientid` is already included in `SkyplaneConfig` [here](https://github.com/skyplane-project/skyplane/blob/f474aaee0203d06497522ef73bd25c93f2512bf3/skyplane/config.py#L158-L160).

**Comment by JasonDing0401** [comment]
This updated version should fix all the issues previously mentioned in the comments. Let me know if you have other suggestions on how to improve this implementation of the opt-out mechanism!

File: skyplane/cli/cli.py (hunk 4 above)
Comment (JasonDing0401) [reply skyplane/cli/cli.py]: moved!

**Comment by parasj** [comment]
@JasonDing0401 I'm getting the following error on `skyplane init`:
```

**Review by parasj — APPROVED** [review]
I pushed a few new changes, but the branch looks good!

**Pull request merged**